pkg/helpers/crd.go

```diff
@@ -102,3 +124,89 @@ func (s SecretSortList) Less(i, j int) bool {
 
 	return s.Items[i].CreationTimestamp.Before(&s.Items[j].CreationTimestamp)
 }
+
+// Checks if the slice of strings contains a specific string
+func SliceContainsString(list []string, s string) bool {
+	for _, v := range list {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+// handleDeletion will handle the finalizers if the object was deleted.
+// It will return true, in case the object has been deleted and the finalizers
+// were removed. If the object was removed the reconcile can be returned.
+func HandleDeletion(instance metav1.Object, resource Remover, finalizerName string, client client.Client) (bool, error) {
+	if instance.GetDeletionTimestamp() == nil {
+		return false, nil
+	}
+
+	annotationValue, exists := instance.GetAnnotations()[DeleteProtectionAnnotation]
+
+	// this is really weird: https://play.golang.org/p/NJkbjN7fCMa
+	var protected bool
+	var err error
+	if exists {
+		protected, err = strconv.ParseBool(annotationValue)
```

Comment:
Will an empty string be true of false?
If I set the annotation `syn.tools/protected-delete=""` I would want it to be true.

Reply:
`syn.tools/protected-delete=""` will throw an error, which is caught and assumed `true` via the if statement on line 154.